include/deal.II/lac/linear_operator.h

```diff
@@ -49,12 +49,7 @@ LinearOperator<Range, Domain> linear_operator (const Matrix &);
 template <typename Domain = Vector<double>,
           typename Range = Domain>
 LinearOperator<Domain, Range>
-null_operator(const std::function<void(Domain &, bool)> &,
-              const std::function<void(Range &,  bool)> &);
-
-template <typename Range = Vector<double>>
-LinearOperator<Range, Range>
-null_operator(const std::function<void(Range &, bool)> &);
+null_operator(const LinearOperator<Domain, Range> &);
```

Comment:
Sure. Consider a rectangular zero block in a 2x2 block system. Regarding the question of different types - we support a linear operator with different domain and range types - so the null operator should also support it.